nemo_aligner/data/nlp/datasets.py

```diff
@@ -344,16 +350,95 @@ def encode(self, text, append_eod=False):
 
         return text_ids, len(text_ids)
 
+    @staticmethod
+    def _convert_messages(
+        input_list: List[Dict[str, str]]
+    ) -> Dict:  # TODO: (@adithyare) this method should live elsewhare..
+        """
+        args:
+            input_list: is a list of dicts in the openai format
+                for example:
+                [{"role": "system", "content": "you are helpful},
+                {"role": "user", "content": "Why is the sky blue?"},
+                {"role": "assistant", "content": "Because blablabla"},
+                ...]
+        returns:
+            output_dict: a dict in nemo's format {"system": "sytem prompt",
+                                                 "conversation": [],
+                                                 ...
+                                                } 
+        """
+        output_dict = {
+            "system": "",
+            "conversations": [],
+            "mask": "User",
+            "type": "VALUE_TO_TEXT",
+        }
+
+        # Extract the system message
+        for msg in input_list:
+            if msg["role"] == "system":
+                output_dict["system"] = msg["content"]
+                break  # Assuming only one system message
```

Comment:
(disclaimer: not very familiar with this format)

Q: is this assumption of having only one system prompt defined in the schema of the openai format? It looks like the format allows for multiple system prompts at least according to this question on their blogs: https://community.openai.com/t/multiple-system-messages/295258

Since it's allowed, should we just throw an error as opposed to silent ignoring the rest?

Reply:
good point, ill throw in an assert -- we should only have one system prompt.